File: android/app/src/main/java/com/autologin/wifi/WifiLoginService.java
package com.autologin.wifi;

import android.app.IntentService;

import android.content.Intent;

import android.util.Log;

import com.autologin.Debug;

public class WifiLoginService extends IntentService {
  public  WifiLoginService() {
    super("WifiLoginService");
  }

  @Override
  protected void onHandleIntent(Intent intent) {
    WifiAccount account = new WifiAccount(this);
    if(!account.isLogin()) {
      Log.i(Debug.TAG, "Service: Not login");
      return;
    }
    LoginWifi login = new LoginWifi(this, account.getUsername(), account.getPassword());
    Log.i(Debug.TAG, "Service: Start login");
    login.login();
  }
}

File: android/app/src/main/kotlin/com/autologin/wifi/WifiLoginService.kt
package com.autologin.wifi

import android.app.IntentService
import android.content.Intent
import android.util.Log
import com.autologin.Debug

class WifiLoginService: IntentService("WifiLoginService") {
  override protected fun onHandleIntent(intent: Intent) {
    val account = WifiAccount(this)
    if (!account.isLogin) {
      Log.i(Debug.TAG, "Service: Not login")
      return
    }
    val login = LoginWifi(this, account.getUsername(), account.getPassword())
    Log.i(Debug.TAG, "Service: Start login")
    login.login()
  }
}
